plugins/Referrers/Columns/Campaign.php

```diff
@@ -50,6 +50,7 @@ public function shouldForceNewVisit(Request $request, Visitor $visitor, Action $
 
         if ($information['referer_type'] == Common::REFERRER_TYPE_CAMPAIGN
             && $this->isReferrerInformationNew($visitor, $information)
+            && !$this->isCurrentReferrerDirectEntry($visitor)
```

Comment:
since it's quite tricky to undersatnd, could you add a comment above this line to explain that eg. "when a visit has referrer = direct entry, and that the current request's referrer = campaign, although you may expect it to create a new visit, actually the referrer will be updated in the current visit (rather than a new visit created.) This is to prevent issue where internal links with URL campaign parameters cause too many visits to be created. " or so

Reply:
done